### PR TITLE
Add support for adding virtual TPM device to VM

### PR DESCRIPTION
**Pull request by Tomorrow9**
Add a new parameter "virtual_tpm", if set to True, will add vTPM device to VM after VM creation.

### CODE REVIEW

File: linux/deploy_vm/add_vm_virtual_tpm.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+#
+# When 'key_provider_type' parameter is set, add a new key
+# provider with specified type firstly, now only support
+# Standard key provider configuration.
+#
+- block:
+    - name: Check key provider type value
+      assert:
+        that:
+          - key_provider_type | lower in ['standard', 'native']
```
Comment (keirazhang) [inline linux/deploy_vm/add_vm_virtual_tpm.yml:15]: I think we can move `linux/deploy_vm/add_vm_virtual_tpm.yml` to `common/vm_add_virtual_tpm.yml` as it doesn't have linux specific operations. And if `native` key provider is not supported at now, we can only check `key_provider_type | lower == 'standard'`. After `native` key provider is supported, we can add it back here.

File: windows/deploy_vm/deploy_vm.yml [hunk 2]
```diff
@@ -71,13 +71,15 @@
...
+        - block:
+            # Add virtual TPM device
+            - include_tasks: ../../linux/deploy_vm/add_vm_virtual_tpm.yml
+              when: virtual_tpm is defined and virtual_tpm | bool
             # Enable secureboot
             - include_tasks: ../../common/vm_set_boot_options.yml
               vars:
                 secure_boot_enabled_set: "{{ secureboot_enabled | default(False) }}"
-                enter_bios_setup: True
+                enter_bios_setup: True
```
Comment (keirazhang) [inline windows/deploy_vm/deploy_vm.yml:82]: The space at the end of this line can be trimmed.

File: env_setup/check_testing_vars.yml [hunk 3]
```diff
@@ -71,4 +71,11 @@
             - cpu_number is defined and cpu_number
             - cpu_cores_per_socket is defined and cpu_cores_per_socket
       when: vm_deploy_method is undefined or vm_deploy_method == "iso"
+
+    - name: "Check vCenter is set when create new VM with vTPM"
+      assert:
+        that:
+          - vcenter_hostname is defined and vcenter_hostname
```
[the remark this answers is not in the document]
Comment (Tomorrow9) [reply env_setup/check_testing_vars.yml:87]: Yeah, that's right. Added vcenter_username, vcenter_password and datacenter here.

File: linux/deploy_vm/add_vm_virtual_tpm.yml (hunk 1 above)
Comment (Tomorrow9) [reply linux/deploy_vm/add_vm_virtual_tpm.yml]: Moved task file to common folder.

**Review by keirazhang — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**